### DISCUSSION TITLE
Stop a simulation when it last too long?

### DISCUSSION THREAD

**Discussion by RuiheLi** [pybamm-team/PyBaMM#2449] [Q&A] [answered]
Dear PyBaMM,

May I ask are there any built-in options for `sim.solve()` which will allow me to stop the simulation when it takes too long (say, 4 hours) and skip to the next lines in the scripts without stopping the whole scripts?

Big thanks in advance!

**Comment by rtimms** [comment, marked as answer]
Nothing built in, but you could wrap the call to solve in some other code. E.g. see [this post](https://stackoverflow.com/questions/492519/timeout-on-a-function-call)

**Reply by RuiheLi** [reply]
Thanks so much! The signal method in the link works! Though signal doesn't not support Windows, but it just means the debug have to move to Linux or other systems.
